### DISCUSSION TITLE
Inplace unattended install?

### DISCUSSION THREAD

**Discussion by Grid21** [memstechtips/UnattendedWinstall#46] [Ideas]
Would it be possible to have an version of Windows 11 upgrade that is for "inplace upgrade" using an unattended.xml file? I've been trying to find information about it on Google and haven't turned up a lot that makes it simpler to do. I'd be happy to help test it in a VM, inplace upgrading from Windows 10 to 11 using an unattended, I am not just not knownledgable yet enough to know how to do it myself. Let me know and thanks!

**Comment by RePacked** [comment]
I asked ChatGPT how you can do this. I am not able to test it right now but you are free to test it yourself.

---

To use an `unattended.xml` file for an in-place upgrade from Windows 10 to Windows 11, you need to follow these steps:

1. **Prepare the Windows 11 Installation Media:**
   - Download the Windows 11 ISO file from the official Microsoft website.
   - Create a bootable USB drive or extract the contents of the ISO to a folder on your computer.

2. **Update the Unattended XML File:**
   - Ensure that your `unattended.xml` file is configured correctly for Windows 11. There might be new or different settings required compared to Windows 10.
   - Make any necessary modifications to the file to account for changes in Windows 11. You might need to refer to the official Microsoft documentation for the specific settings applicable to Windows 11.

3. **Place the Unattended XML File:**
   - Copy the `unattended.xml` file to the `sources` folder of your Windows 11 installation media. The path should be: `path-to-windows-11-media\sources\unattend.xml`.

4. **Run the Upgrade:**
   - Open a Command Prompt with administrative privileges.
   - Navigate to the folder containing the Windows 11 installation files.
   - Run the setup with the following command:

     ```cmd
     setup.exe /auto upgrade /unattend:path-to-unattend.xml
     ```

Replace `path-to-unattend.xml` with the actual path to your `unattended.xml` file if it is not in the `sources` folder.

This command will start the in-place upgrade process using the settings specified in the `unattended.xml` file.

**Reply by Grid21** [reply]
Cool! I will try this in a VM and let you know the results! This project for doing unattended installs is SUPER cool and I am ALL HERE for it! So if I can test stuff for the project in a VM that's fine with me!

**Reply by Grid21** [reply]
Tes Results came back, my full command line was E:\>setup.exe /Auto:Upgrade /unattend:E:\sources\unattend.xml Windows 10 replied with error "An Unknown command-line option [/Auto:Upgrade] was specified. Apparently this wasn't correct so I am confused why ChatGPT thought that was the correct command. 

EDIT: turns out the correct command was this: setup.exe /unattend:E:\sources\unattend.xml. Rerunning the tests now with a custom unattend file.

Updated EDIT: The command worked, but unfornately, doing the upgrade ends up wiping the computer clean, instead of keep existing user and program data. So what can be done on the website that lets you customize your xml to keep your data intact?

**Comment by memstechtips** [comment]
Unfortunately, you can't use it with an in place upgrade and keep all of your files, it basically does a clean install.

However I have released a new version of the XML file and a major update to [Winhance](https://winhance.net/), I recommend that you watch [this video](https://youtu.be/lrq3ph3xi50) which provides a full demo of the current version. You can just import and apply the recommended config file after doing an in place upgrade to a new version of Windows and it will give you the same experience as UnattendedWinstall.

I have also created [this video](https://youtu.be/sCd1TI6AbOI) about v3.0.0 of UnattendedWinstall, I suggest that you watch it too if you're interested.
